feat(send): add unified onchain/lightning switch

pwltr opened the pull request:
### Description

- Add ability to switch between sending from savings & spending for 0 amount unified QRs. For this I added a new field in the `ui` reducer `paymentMethod` that is set in `handleData` (last step of `processUri`) and read in the send flow. There is also a new QrType "Unified"
- Refactor scanning logic to avoid duplication and improve readability. The goal is to have clear steps for processing any URI we support: 1. parse 2. validate 3. process/transform for handling 4. actually handle (UI actions)
- Add e2e tests in new file `send.e2e.js`. I would like to have this be the main file for testing send flow since other tests are concerned with only onchain or lightning at the moment, but edge cases need to be handled for all variants.

### Linked Issues/Tasks

Closes https://github.com/synonymdev/bitkit/issues/2003

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (improving code without creating new functionality)

### Tests

- [x] Detox test
- [ ] Unit test
- [ ] No test

### Screenshot / Video

https://github.com/user-attachments/assets/350d06ea-f895-4519-b75f-9913a037e231

### QA Notes

I tested basic functionality as well as max amount, coin selection and slashpay.

pwltr commented on e2e/helpers.js:
```diff
@@ -47,22 +47,9 @@ export const sleep = (ms) => {
-		return true;
-	} catch (e) {
-		return false;
-	}
```
These didn't work or broke a while back. New one is reliable.

limpbrains commented on src/utils/lnurl.ts:
```diff
@@ -221,15 +220,15 @@ export const handleLnurlChannel = async ({
 			: i18n.t('other:lnurl_channel_success_msg_no_peer'),
 	});
-	return ok({ type: EQRDataType.lnurlChannel });
+	return ok('');
```
OK :)

pwltr replied:
Just to clarify the `ok` result was not used anywhere, this makes it clear we don't have to care for this return value.

limpbrains replied:
agree

limpbrains commented:
Unit tests are failing

limpbrains commented:
When my wallet has no LN channels and I scan LN+onchain qrcode, I get this error:
<img width="200" src="https://github.com/user-attachments/assets/1d381a24-c231-4bd6-a5c2-d12b59a5eb10" />

Should it be so?

limpbrains commented:
I aslo have this error if I have 0 Onchain balance and scanning QRcode with Onchain address only

<img width="200" src="https://github.com/user-attachments/assets/14f6b17d-df39-42b8-9d07-e45b54b39e87" />

pwltr commented:
> When my wallet has no LN channels and I scan LN+onchain qrcode, I get this error: <img alt="" width="200" src="https://private-user-images.githubusercontent.com/155891/382749874-1d381a24-c231-4bd6-a5c2-d12b59a5eb10.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA3MjMwNDcsIm5iZiI6MTczMDcyMjc0NywicGF0aCI6Ii8xNTU4OTEvMzgyNzQ5ODc0LTFkMzgxYTI0LWMyMzEtNGJkNi1hNWMyLWQxMmI1OWE1ZWIxMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNFQxMjE5MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNTQ4MDVjZTQ5YWZmMzc5YjY0ZjIwODg3NWRjMjQzNTg5MThlNzNmYTZlY2Q4MjcyZTQ2OGQ2NDY0OTJiODY0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.D8bnrNaqLAWL3UnkW5LMivLSdwBIDAzphahfdPJIm9I">
> 
> Should it be so?

Looks like this one is a regression.

> I aslo have similar error if I have 0 Onchain balance and scanning QRcode with Onchain address only
> 
> <img alt="" width="200" src="https://private-user-images.githubusercontent.com/155891/382750995-14f6b17d-df39-42b8-9d07-e45b54b39e87.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA3MjMwOTIsIm5iZiI6MTczMDcyMjc5MiwicGF0aCI6Ii8xNTU4OTEvMzgyNzUwOTk1LTE0ZjZiMTdkLWRmMzktNDJiOC05ZDA3LWU0NWI1NGIzOWU4Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNFQxMjE5NTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zM2IxZjI4MDhjZTVmOTY4ZTRlMTM3NmFmZWJiY2IxYjEzNmIwMzRmYTVjNzM5YTU1ZDRmNDBmODA2MDhmNzIzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.3jPkIsWekUtH2dwgp6R3SanHsX-UXBR8jxLhedzBnjk">

This one is expected behavior.

limpbrains commented:
Tapping on Available balance when trying to send from Spending account works only once
And stops working after switching to Savings and back

https://github.com/user-attachments/assets/f5ee62f9-4a30-4cc0-94c9-12f3cfbd58e6

limpbrains commented:
When scan Unified QR code with onchain and LN amount exceeding your balance, bitkit opens Spending and allows me to specify the amount. And then fails.

https://github.com/user-attachments/assets/9f7bfe9d-ae6a-4005-91c1-3895b427f248

I actually can switch to Savings in this case. 
IDK what should happend in this case, maybe just an error and nothing else?

limpbrains commented:
Sometimes Bitkit doesn't allow me to choose how to send. 
LN is forced in this example, while I have funds to pay via LN or on-chain.

https://github.com/user-attachments/assets/42d832c4-bd21-41fe-9884-c9accbcc8594

limpbrains commented:
> This one is expected behavior.

Error message doesn't make sense

limpbrains commented:
After scanning the Unified QR without an amount and switching to Savings, I get the Spending MAX amount when I tap on Available balance.

https://github.com/user-attachments/assets/19c5c0d3-f192-474a-b59c-533d5fa4b26d

pwltr commented:
> Sometimes Bitkit doesn't allow me to choose how to send. LN is forced in this example, while I have funds to pay via LN or on-chain.
> 
>  kek2.mov

This feature is only for 0 amount unified invoices. See description at the top.

pwltr commented:
> Tapping on Available balance when trying to send from Spending account works only once And stops working after switching to Savings and back

Fixed and added tests

pwltr commented:
> > This one is expected behavior.
> 
> Error message doesn't make sense

Fixed. Toast says "More ₿ needed to pay this Bitcoin invoice." now like before.

pwltr commented:
Think I fixed everything, hopefully it's good now.
